doc/storage_plugins.md

```diff
@@ -4,20 +4,183 @@ Storage plugins
 
+DLite provide a plugin system that make it easy to connect to new data sources via a common interface (using a [strategy design pattern]).
+Opening a storage takes three arguments, a `driver` name identifying the storage plugin to use, the `location` of the storage and `options`.
 
-Using Python
-------------
-xxx
+Storage plugins can be categorised as either *generic* or *specific*.
+A generic storage plugin is can store and retrieve any type of instances and metadata while specific storage plugin is typically specific instances of one type of entity.
+DLite comes with a set of generic storage plugins, like json, yaml, rdf, hdf5, postgresql and mongodb.
+It also comes with a specific Blob storage plugin, that can load and save instances of a `http://onto-ns.com/meta/0.1/Blob` entity.
+Storage plugins can be written in either C or Python.
 
 
-Writing in C
-------------
-xxx
+Working with storages in Python
+-------------------------------
+For instance, to create a new file-based JSON storage can in Python be done with:
+
+```python
+    >>> import dlite
+    >>> s = dlite.Storage("json", location="newfile.json", options="mode=w")
+
+```
+
+where the first argument is the name of the driver to use.
+It is also possible to combine the three arguments into an URL as in
+
+```python
+    >>> s = dlite.Storage("json://newfile.json?mode=w")
+
+```
+
+Note that the URL form less flexible, since the scheme part identifies the driver and can therefore not be used in the location of an external storage.
+
+The options are plugin specific, but most file-based plugins take a `mode` option, with the following three values:
+- `r`: Open existing file for read-only.
+- `w`: Truncate existing file or create new file.
+- `a`: Append to existing file or create new file.
+
+The values of `driver`, `location` and `options` are available as attributes:
+
+```python
+    >>> s.driver
+    'json'
+    >>> s.location
+    'newfile.json'
+    >>> s.options
+    'mode=w'
+
+```
+
+Currently, the Python interface does not implement a `close()` method to avoid risking unexpected crashes if one tries to access a closed storage object.
+If you want to ensure a storage is closed (and possible buffered data is committed), you can delete it:
+
+```python
+    >>> del s
+
+```
+
+Alternatively you can open a storage is in a `with`-statement.
+This is the preferred way, since it ensures that the storage
+
+```python
+    >>> with dlite.Storage("json", "newfile.json", options="mode=w") as s:
+    ...     ...  # doctest: +SKIP
+
+```
+
+Instances can be stored using the `save()` method:
+
+
+```python
+    # Create some Blob data instances
+    >>> Blob = dlite.get_instance('http://onto-ns.com/meta/0.1/Blob')
+    >>> blob1 = Blob(dims={"n": 2}, id="ex:blob1")
+    >>> blob2 = Blob(dims={"n": 5}, id="ex:blob2")
+
+    # Add the instances to the storage
+    >>> with dlite.Storage("json", location="newfile.json",
+    ...                    options="mode=w;single=no") as s:
```

Comment:
What does the "single" option do? Would be great to have a comment about this since it has not been mentioned earlier in the text.

Reply:
La inn kommentarer direkte i kode-eksemplet.